### PR TITLE
chore(httpPost): Adding http post method in httpProbe

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Adding http post method inside the http probe
- Adding criteria match for the response code for HTTP Get request
- changes needed for PR: litmuschaos/litmus-go#281

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Labelled this PR & related issue with `documentation` tag
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests

### CODE REVIEW

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 1]
```diff
@@ -267,8 +267,30 @@ type HTTPProbeInputs struct {
...
+type GetMethod struct {
+	// Criteria for matching data
+	// it supports  == != operations
+	Criteria string `json:"criteria,omitempty"`
+	// Value contains relative value for criteria
+	ResponseCode string `json:"responseCode,omitempty"`
+}
+
+// PostMethod define the http Post method
+type PostMethod struct {
```
Comment (ksatchit) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:290]: Do we need a response code validation here as well?

**Review by ksatchit — APPROVED** [review]

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 2]
```diff
@@ -267,8 +267,38 @@ type HTTPProbeInputs struct {
...
-	ExpectedResponseCode string `json:"expectedResponseCode,omitempty"`
+	// Method define the http method, it can be get or post
+	Method HTTPMethod `json:"method,omitempty"`
+}
+
+// HTTPMethod define the http method details
+type HTTPMethod struct {
+	Get  GetMethod  `json:"get,omitempty"`
+	Post PostMethod `json:"post,omitempty"`
+}
```
Comment (rahulchheda) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:278]:
```suggestion
	ResponseCode string `json:"responseCode,omitempty"`
}
```

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 3]
```diff
@@ -267,8 +267,38 @@ type HTTPProbeInputs struct {
...
+
+// GetMethod define the http Get method
+type GetMethod struct {
+	// Criteria for matching data
+	// it supports  == != operations
+	Criteria string `json:"criteria,omitempty"`
+	// Value contains relative value for criteria
+	ResponseCode string `json:"responseCode,omitempty"`
```
Comment (rahulchheda) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:286]:
```suggestion
```

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 4]
```diff
@@ -267,8 +267,38 @@ type HTTPProbeInputs struct {
...
+	ContentType string `json:"contentType,omitempty"`
+	// Body contains http body for post request
+	Body string `json:"body,omitempty"`
+	// BodyPath contains filePath, which contains http body
+	BodyPath string `json:"bodyPath,omitempty"`
+	// Criteria for matching data
+	// it supports  == != operations
+	Criteria string `json:"criteria,omitempty"`
+	// Value contains relative value for criteria
+	ResponseCode string `json:"responseCode,omitempty"`
```
Comment (rahulchheda) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:301]:
```suggestion
```

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go (hunk 2 above)
Comment (rahulchheda) [reply pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:278]: Common fields can exist in this struct.

**Review by rahulchheda — APPROVED** [review]
nit

**Pull request merged**